pkg/ibmcsidriver/node.go

```diff
@@ -148,6 +144,11 @@ func (csiNS *CSINodeServer) NodePublishVolume(ctx context.Context, req *csi.Node
 	var nodePublishResponse *csi.NodePublishVolumeResponse
 	var mountErr error
 
+	//Lets try to put lock at targetPath level. If we are processing same target path lets wait for other to finish.
+	//This will not hold other volumes and target path processing.
+	csiNS.mutex.Lock(target)
```

Comment:
how many parallel volume mount tried on same node? and see how resource utilisation

Reply:
Please refer this GHE for more details we have tried parallel 30 to 50 mounts.

https://github.ibm.com/alchemy-containers/armada-storage/issues/5245#issuecomment-69052970